test/ValueGetterTest.cpp

```diff
@@ -0,0 +1,34 @@
+//  Copyright 2022, University of Freiburg,
+//  Chair of Algorithms and Data Structures.
+//  Author:
+
+#include <gtest/gtest.h>
+
+#include "./SparqlExpressionTestHelpers.h"
+#include "engine/sparqlExpressions/SparqlExpressionValueGetters.h"
+
+TEST(LiteralValueGetter, OperatorWithId) {
+  sparqlExpression::TestContext testContext{};
+  sparqlExpression::detail::LiteralValueGetter literalValueGetter;
+  Id idLiteral = testContext().alpha; //literal with no datatype
+  auto literal = literalValueGetter(idLiteral, &testContext.context);
+  ASSERT_TRUE(literal.has_value());
+  ASSERT_EQ(literal.value().getContent(), "alpha");
+  //TODO: Test Ids that hold a iri.
```

Comment:
Is your question how go get Ids for certain values/IRIs/Ids?

You can create a test index by `ad_utility::testing::getQec(someTurtle)` make sure that the  turtle contains all the things you want, for example
```
PREFIX xsd: #Insert correct prefix here
<x> <y> "alpha"^^xsd:string.
<x> <y> "blubb"^^<someStrangeDatatype>.
...
 ```

Then you can use the `makeGetId` helper (see in the tests), to get the correct ID for the entries, and then test that.